hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/NewHoodieParquetFileFormat.scala

```diff
@@ -74,11 +76,25 @@ class NewHoodieParquetFileFormat(tableState: Broadcast[HoodieTableState],
   override def supportBatch(sparkSession: SparkSession, schema: StructType): Boolean = {
     if (!supportBatchCalled) {
       supportBatchCalled = true
-      supportBatchResult = !isMOR && super.supportBatch(sparkSession, schema)
+      supportBatchResult = !isMOR && legacyFF.supportBatch(sparkSession, schema)
     }
     supportBatchResult
   }
 
+  private def wrapWithBatchConverter(reader: PartitionedFile => Iterator[InternalRow]): PartitionedFile => Iterator[InternalRow] = {
```

Comment:
why is this needed? i think the flatmap per row could incur some significant cost for a large batch. Instrad of wrapping, can it be guarded for some cases such as when schema on read is enabled?

Reply:
Right.  @jonvex I think Spark internally handles the batch processing (`InternalRow` vs `ColumnarBatch`) based on the boolean `supportBatch` returns.  So we don't have to do batch converter here?